bin/kano-settings-onboot

```diff
@@ -135,27 +158,122 @@ def compare_and_set_overscan():
         return True
 
 
-def get_screen_information():
-    """ Retrieves the information about the current screen.
+def compare_and_set_optimal_resolution():
+    '''
+    This function tries to set the best performance resolution mode on the kit.
+    It tries to find the preferred screen mode and aspect ratio, to try and match
+    an optimal resolution. If all fails, a default resolution is still tried.
+
+    The end result is to stop having the Raspberry run at the highest possible resolution,
+    which slows down a number of graphical heavyweight applications.
+
+    On success, the new mode is enforced (hdmi_group / hdmi_mode) and True is returned.
+    '''
+    global edid, status, supported, verbose
+    preferred_mode = preferred_group = preferred_aspect_ratio = optimal_mode = None
+
+    logger.info('Searching for the best optimal resolution')
+
+    def find_best_resolution_aspect(aspect, width, height):
+        '''
+        Finds the screen mode at the given resolution and aspect ratio,
+        If aspect is None, the first matching resolution mode will be returned, if found.
+        '''
+        for mode in supported:
+            w = int(mode['resolution'].split('x')[0])
+            h = int(mode['resolution'].split('x')[1])
```

Comment:
Maybe abstract this to the parser so that we have something along the lines of

```python
mode['resolution'] = {
    'width': <resolution.split('x')[0]>,
    'height': <resolution.split('x')[1]>
}
```